src/coaster_gui.cpp

```diff
@@ -161,6 +164,11 @@ void CoasterInstanceWindow::OnClick(WidgetNumber widget, const Point16 &pos)
 		case CIW_OPEN:
 			this->SetCoasterStateForRadioButton(widget);
 			break;
+
+		case CIW_EDIT_NAME:
+			if (HighlightWindowByType(WC_EDIT_TEXT, ALL_WINDOWS_OF_TYPE)) return;
+			ShowEditTextGui(&this->ci->name, MAX_RIDE_INSTANCE_NAME_LENGTH);
```

Comment:
This looks horribly scary, a first search in the code suggests the text isn't actually copied anywhere, which means it's editing the memory block owned by the coaster. This is also somewhat confirmed by lack of a parameter so you can find the coaster instance again when editing is finished.
If this is indeed true, the problem with this solution is that you can create all kinds of nice crashes by opening a edit session and then delete the coaster instance, so you're editing unowned memory.

It has been too long ago to know if the above holds.